impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -153,363 +97,168 @@ public synchronized void stop() {
-    } else {
-      boolean deleteSuccessful = true;
-      for (File f : contents) {
-        deleteSuccessful &= tryRecursiveDelete(f);
-      }
-      return deleteSuccessful;
-    }
-  }
-
-  private static boolean recursiveDelete(File file) {
-    Deque<File> toDelete = new ArrayDeque<File>();
-    toDelete.push(file);
-    while (!toDelete.isEmpty()) {
-      File target = toDelete.pop();
-      File[] contents = target.listFiles();
-      if (contents == null || contents.length == 0) {
-        if (target.exists() && !target.delete()) {
-          return false;
-        }
-      } else {
-        toDelete.push(target);
-        for (File f : contents) {
-          toDelete.push(f);
-        }
-      }
-    }
-    return true;
-  }
-
-  @SuppressFBWarnings("DM_GC")
-  private static boolean tryRecursiveDelete(File file) {
-    boolean interrupted = false;
-    try {
-      for (int i = 0; i < 5; i++) {
-        if (recursiveDelete(file) || !isWindows()) {
-          return true;
-        } else {
-          System.gc();
-          System.runFinalization();
-
-          try {
-            Thread.sleep(50);
-          } catch (InterruptedException e) {
-            interrupted = true;
-          }
-        }
-      }
-    } finally {
-      if (interrupted) {
-        Thread.currentThread().interrupt();
-      }
-    }
-    return false;
- }
-
-  private static boolean isWindows() {
-    return System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("windows");
-  }
-
-  /**
-   * sanitize a name for valid file or directory name
-   *
-   * @param name the name to sanitize
-   * @return sanitized version of name
-   */
-  private static String safeIdentifier(String name) {
-    return safeIdentifier(name, true);
-  }
-
-  static String safeIdentifier(String name, boolean withSha1) {
-    int len = name.length();
-    StringBuilder sb = new StringBuilder(len);
-    for (int i = 0; i < len; i++) {
-      char c = name.charAt(i);
-      if (c <= ' ' || c >= DEL || ILLEGALS.contains(c) || c == ESCAPE) {
-        sb.append(ESCAPE);
-        sb.append(String.format("%04x", (int) c));
-      } else {
-        sb.append(c);
-      }
-    }
-    if (withSha1) {
-      sb.append("_").append(sha1(name));
-    }
-    return sb.toString();
-  }
-
-  private static String sha1(String input) {
-    StringBuilder sb = new StringBuilder();
-    for (byte b : getSha1Digest().digest(input.getBytes(UTF8))) {
-      sb.append(toHexString((b & 0xf0) >>> 4));
-      sb.append(toHexString((b & 0xf)));
-    }
-    return sb.toString();
-  }
 
-  private static MessageDigest getSha1Digest() {
-    try {
-      return MessageDigest.getInstance("SHA-1");
-    } catch (NoSuchAlgorithmException e) {
-      throw new AssertionError("All JDKs must have SHA-1");
-    }
+  private SafeSpace createSafeSpaceLogical(String owner, String identifier) {
+    File ownerDirectory = new File(rootDirectory, owner);
```

Comment:
Yes. owner is typically hardcoded, pre-defined values used by Ehcache persistable resource services only. It is to identify a root for a given service that is separate from other services,